src/main/protocol.js

```diff
@@ -1,19 +1,63 @@
 const {
   protocol,
+  session,
 } = require('electron');
 const {
   readFileSync,
 } = require('node:fs');
 const url = require('node:url');
+const { refreshJit } = require('./containers/refresh-jit');
 
+function registerSchemas() {
+  protocol.registerSchemesAsPrivileged([{
+    scheme: 'jit',
+    privileges: { supportFetchAPI: true },
+  }]);
+}
 function registerHandlers() {
   protocol.handle('awsaml', (request) => {
     const prefix = 'awsaml://'.length;
 
     return new Response(readFileSync(url.fileURLToPath(`file://${request.url.slice(prefix)}`)));
   });
+
+  protocol.handle('jit', async (request) => {
+    let sessionId;
+    const { host, pathname } = new URL(request.url);
+    if (host === 'get-active-profiles') {
+      const activeProfiles = Object.values(Manager.reloaders).map((r) => r.role);
+      return new Response(JSON.stringify({ activeProfiles }));
+    }
+    return session.defaultSession.cookies.get({})
```

Comment:
You can consolidate this call and the `cookies.find` call a few lines down by passing a filter object to the [`get` method](https://www.electronjs.org/docs/latest/api/cookies#cookiesgetfilter). If you're looking for the `session_id` cookie you could do:
```suggestion
    return session.defaultSession.cookies.get({
    	name: 'session_id',
    })
```
One thing to note is that `Array.find` used below returns the first element that matches the predicate regardless of whether more matches exist. I'd suggest trying to figure out an additional way of identifying the correct `session_id` cookie from ICS. Maybe domain matching or something?